src/raft/raft.go

```diff
@@ -382,6 +384,15 @@ func (rf *Raft) sendRequestVote(server int, args *RequestVoteArgs, reply *Reques
 					}
 				}
 				rf.mu.Unlock()
+			} else {
+				// update myself if reply.Term > rf.currentTerm; make sure i am still a candidate
+				rf.mu.Lock()
+				if reply.Term > rf.currentTerm && rf.state == Candidate {
```

Comment:
This fixes this scenario in `TestBackup2B`:
1. Let's say we start with 5 servers, A,B,C,D,E
2. A is leader; we make 1 commit
3. We disconnect C,D,E
4. We send A some commands, these will be replicated on B, but will NOT be committed
5. Then we disconnect A,B; reconnect C,D,E; assume C is the new leader with new term
6. We make some commits to the new group of C,D,E
7. Note that the disconnected B will transition into candidate, because B won't hear any heart beat from A as it's disconnected; 
8. Then we disconnect D; send some more commands to C, these will be replicated to E, but will NOT be committed
9. Then we make a group of A,B,D; Note D is the only legitimate leader as it has commits with higher terms
However, B will have advanced a few terms by now, and as A will quickly get same term from B from a failed heartbeat reply; as a result, both A and B will likely be a few terms ahead of D
10. Then we rely on randomised election timeout to wait for D to catch up eventually; and this takes time.